src/facade/dragonfly_connection.cc

```diff
@@ -655,19 +668,23 @@ auto Connection::IoLoop(util::FiberSocketBase* peer) -> variant<error_code, Pars
 
 struct Connection::DispatchOperations {
   DispatchOperations(SinkReplyBuilder* b, Connection* me)
-      : stats{me->service_->GetThreadLocalConnectionStats()}, builder{b},
-        empty{me->dispatch_q_.empty()}, self(me) {
+      : stats{me->service_->GetThreadLocalConnectionStats()}, builder{b}, self(me) {
   }
 
   void operator()(PubMsgRecord& msg);
   void operator()(Request::PipelineMsg& msg);
+  void operator()(const Request::MonitorMessage& msg);
 
   ConnectionStats* stats = nullptr;
   SinkReplyBuilder* builder = nullptr;
-  bool empty = false;
   Connection* self = nullptr;
 };
 
+void Connection::DispatchOperations::operator()(const Request::MonitorMessage& msg) {
+  RedisReplyBuilder* rbuilder = (RedisReplyBuilder*)builder;
+  rbuilder->SendSimpleString(msg);
```

Comment:
Does Redis sends simple messages in that case?

Reply:
yes - they are building a string and sending it, I actually "copy" some of their code (with modifications to C++) to be the same as they are see for example CmdEntryToMonitorFormat function in main_service.cc